RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,275 @@
+```
+
+There are two aspects of making this work well:
+
+- Knowing that the the parameter `--format='{{json .}}` is available for the `docker` command.
+- Knowing that the invocation will emit json and then suggest how to convert that to objects.
+
+### Providing feedback to the user about improved pipelines
+
+We can improve the experience by using the `FeedbackProvider` infrastructure and present feedback to the user which provides an improved pipeline.
+
+For example, if the user types:
+
+```powershell
+PS> docker image ls
+```
+
+We can provide the following feedback to the user:
+
+```powershell
+docker image ls --format='{{json .}}' | ConvertFrom-Json
+```
+
+or if the user types the following:
+
+```powershell
+PS> uname -a
+```
+
+and the `jc` utility is available, we can provide feedback with an improved pipeline:
+
+```powershell
+PS> uname -a | jc --uname | ConvertFrom-Json
+```
+
+If the user types:
+
+```powershell
+PS> ipconfig
+```
+
+and if a script which converts the output of `ipconfig` into PowerShell objects named `ipconfig-json` exists, we could suggest the following:
+
+```powershell
+PS> ipconfig | ipconfig-json
+```
+
+This last suggested pipeline could be generalized to include a suggestion for any `<utility>-json` command if found.
+This may encourage the community to create conversion functions/scripts.
+
+#### Needed enhancement to current feedback system
+
+Today, this behavior is not possible because the feedback provider will only act if there was an error in execution.
+To support this, the feedback subsystem would need to be extended so it may be called when no error occurred,
+see [issue19372](https://github.com/PowerShell/PowerShell/issues/19372).
+
+### Open Questions
+
+- How should errors during conversion be handled?
+  - In the case where ConvertFrom-Json is used, I would suggest a new parameter for `ConvertFrom-Json` where it simply emits the input string if the conversion fails.
+
+### Follow-on Work
+
+A number of follow-on activities should be considered to bring continued improvement of the native command experience.
+
+#### Conversion of Tabular Text
+
+If the data is _known_ to be tabular it may be possible to provide an additional utility to convert that tabular data to
+structured data via means of a generalized converter.
+This would require some sort of registration/configuration to designate the native application and call the
+converter to automatically construct objects from tabular data could be called.
+This would enable the user to set it once and later sessions would be able to use it.
+This can be done in a stand-alone module or incorporated in the the TextUtility module in the gallery.
+It is reasonable to do this manually, or perhaps added to the feedback provider suggestion above.
+
+#### PSDefaultNativeParameterValues
+
+We can extend the _concept_ of `$PSDefaultParameterValue` to improve the user experience by automatically
```

Comment:
would it be too complex for the feedback provider to check if there is a default param value on some of its known ones and then not make suggestions? Just wondering about clobbering behaviors